### PR TITLE
Allow unsafe raw SQL injection in PostgresQuery

### PR DESCRIPTION
**Pull request by fabianfett**
### Motivation

Sometimes developers want to have the freedom to inject values into a PostgresQuery that are not bindable. Examples for this can be tableNames or columnNames.

### Changes

- Allow adopters to interpolate raw sql into the PostgresQuery

### Result

More use-cases covered. Make unsafe code explicit unsafe.

### Open question

What is the best unsafe marker word? Right now I have `unsafeSQLInjection`. I like this because it makes potential users google for SQL injection... This is rather long. Other previously discussed options are: `raw`, `unsafeRaw`.

### CODE REVIEW

File: Sources/PostgresNIO/New/PostgresQuery.swift [hunk 1]
```diff
@@ -67,6 +67,11 @@ extension PostgresQuery {
             try self.binds.append(value, context: context)
             self.sql.append(contentsOf: "$\(self.binds.count)")
         }
+
+        @inlinable
+        public mutating func appendInterpolation(unsafeSQLInjection interpolated: String) {
```
Comment (gwynne) [inline Sources/PostgresNIO/New/PostgresQuery.swift:72]: I think the word `Injection` is overkill in this case - `unsafe` is at least theoretically supposed to be a sufficiently big signal flare in _any_ Swift code to cover it. I'd tend towards making the label `unsafeSQL` and then clearly defining SQL injection and its dangers (in short form, not one of my rants!! 😆) as a documentation comment.

File: Tests/PostgresNIOTests/New/PostgresQueryTests.swift [hunk 2]
```diff
@@ -79,4 +76,19 @@ final class PostgresQueryTests: XCTestCase {
 
         XCTAssertEqual(query.binds.bytes, expected)
     }
+
+    func testUnsafeSQLInjection() {
```
Comment (gwynne) [inline Tests/PostgresNIOTests/New/PostgresQueryTests.swift:80]: This is just me being silly, but you know, if you're gonna test SQL injection, why not at least have the test do a classical injection? Say, `#"SELECT foo FROM dual LIMIT 0; INSERT INTO users (username, password, isSuperuserOverAllTheInternet) VALUES ("Good ol' Bobby Tables", "easy password", 1); -- PWNED"#` 😇

**Review by gwynne — CHANGES_REQUESTED** [review]

File: Sources/PostgresNIO/New/PostgresQuery.swift (hunk 1 above)
Comment (fabianfett) [reply Sources/PostgresNIO/New/PostgresQuery.swift:72]: However unsafeSQL doesn't give users such clear google results. SQL injection results are very clear...

Comment (0xTim) [reply Sources/PostgresNIO/New/PostgresQuery.swift:72]: Yeah I agree `unsafeSQLInjection` is confusing. We can add it as a comment if we're concerned about searching. I like `\(unsafeRaw: "")` or `\(raw: "")` - lines up with the rest of the cases across Vapor where we allow unsafe injection of variables and still makes it clear

Comment (adam-fowler) [reply Sources/PostgresNIO/New/PostgresQuery.swift:72]: I think the full name is too wordy. Surely we can detail the issues with sql injection in the comments. I would go further than @gwynne i don't think SQL is required. It should be obvious it is SQL as it is inside a SQL query. So `\(unsafe: ...)` or `\(unsafeRaw...)` should be enough.

Comment (Jerry-Carter) [reply Sources/PostgresNIO/New/PostgresQuery.swift:72]: I’ll put in a vote for ‘unsafeRaw’ in that it suggests direct SQL phrases without escaping or other processing for safety.

Comment (fabianfett) [reply Sources/PostgresNIO/New/PostgresQuery.swift:72]: I talked to @weissi about this yesterday, and we came up with `unescaped`, which I like the most so far. @0xTim, @gwynne, @adam-fowler, @Jerry-Carter What do you think about this?

Comment (0xTim) [reply Sources/PostgresNIO/New/PostgresQuery.swift:72]: Yeah I'm happy with that

Comment (adam-fowler) [reply Sources/PostgresNIO/New/PostgresQuery.swift:72]: Sounds good

**Review by 0xTim — APPROVED** [review]
LGTM

File: Sources/PostgresNIO/New/PostgresQuery.swift (hunk 1 above)
Comment (gwynne) [reply Sources/PostgresNIO/New/PostgresQuery.swift]: I want to argue for `unsafeUnescaped`, but not enough to die on the hill; I can live with this 🙂

**Review by gwynne — APPROVED** [review]
